src/components/dashjs-statistics/dashjs-statistics.tsx

```diff
@@ -89,122 +83,48 @@ export class DashjsStatistics {
-            videoHttpMetrics.download['video'].high.toFixed(2),
-        );
-        this.videoMetricsDataMap['Latency'].push(
-          videoHttpMetrics.latency['video'].low.toFixed(2) +
-            ' | ' +
-            videoHttpMetrics.latency['video'].average.toFixed(2) +
-            ' | ' +
-            videoHttpMetrics.latency['video'].high.toFixed(2),
-        );
-        this.videoMetricsDataMap['Ratio'].push(
-          videoHttpMetrics.ratio['video'].low.toFixed(2) + ' | ' + videoHttpMetrics.ratio['video'].average.toFixed(2) + ' | ' + videoHttpMetrics.ratio['video'].high.toFixed(2),
-        );
-      }
-
-      // Audio Metrics
-      const audioRepSwitch = dashMetrics?.getCurrentRepresentationSwitch('audio');
-      const audioHttpMetrics = calculateHTTPMetrics('audio', dashMetrics?.getHttpRequests('audio'));
-
-      this.audioMetricsDataMap['Buffer Length'].push(dashMetrics?.getCurrentBufferLevel('audio'));
-      this.audioMetricsDataMap['Dropped Frames'].push(dashMetrics?.getCurrentDroppedFrames('audio')?.droppedFrames);
-      this.audioMetricsDataMap['Bitrate Downloading'].push(audioRepSwitch ? Math.round(dashAdapter?.getBandwidthForRepresentation(audioRepSwitch.to, periodIdx) / 1000) : NaN);
-      this.audioMetricsDataMap['Max Index'].push(dashAdapter?.getMaxIndexForBufferType('audio', periodIdx));
-      if (audioHttpMetrics) {
-        this.audioMetricsDataMap['Download'].push(
-          audioHttpMetrics.download['audio'].low.toFixed(2) +
-            ' | ' +
-            audioHttpMetrics.download['audio'].average.toFixed(2) +
-            ' | ' +
-            audioHttpMetrics.download['audio'].high.toFixed(2),
-        );
-        this.audioMetricsDataMap['Latency'].push(
-          audioHttpMetrics.latency['audio'].low.toFixed(2) +
-            ' | ' +
-            audioHttpMetrics.latency['audio'].average.toFixed(2) +
-            ' | ' +
-            audioHttpMetrics.latency['audio'].high.toFixed(2),
-        );
-        this.audioMetricsDataMap['Ratio'].push(
-          audioHttpMetrics.ratio['audio'].low.toFixed(2) + ' | ' + audioHttpMetrics.ratio['audio'].average.toFixed(2) + ' | ' + audioHttpMetrics.ratio['audio'].high.toFixed(2),
-        );
-      }
-    }
+  private currentMetric(metrics: any, type: string) {
+    return (
+      <ion-row>
+        <ion-col size="12">
+          <ion-title>{type.charAt(0).toUpperCase() + type.slice(1)}</ion-title>
+          {Object.keys(metrics[type]).map(metric => (
+            <ion-item class="inline-item">
+              <ion-label class="rmv-b-border">
+                {metric}: {metrics[type][metric]}
+              </ion-label>
+              <ion-checkbox checked color="primary" slot="start" onIonChange={ev => this.chartVisibility(true, [metric], ev.detail.checked)}></ion-checkbox>
+            </ion-item>
+          ))}
+        </ion-col>
+      </ion-row>
+    );
   }
 
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
why? you only have to add a return type?

Reply:
but we can add it to the eslint file if you like

Reply:
it needs to put void for componentDidLoad(), is this appropriate?